packages/pigeon/run_tests.sh

```diff
@@ -86,195 +90,337 @@ test_pigeon_android() {
     --java_out $temp_dir/Pigeon.java \
     --java_package foo
 
-  java -jar ci/$JAVA_FORMATTER --replace "$temp_dir/Pigeon.java"
-  java -jar ci/$JAVA_LINTER -c "ci/$GOOGLE_CHECKS" "$temp_dir/Pigeon.java"
+  java -jar ci/$java_formatter --replace "$temp_dir/Pigeon.java"
+  java -jar ci/$java_linter -c "ci/$google_checks" "$temp_dir/Pigeon.java"
   if ! javac \
-    $JAVAC_BOOTCLASSPATH \
+    $javac_bootclasspath \
     -XDcompilePolicy=simple \
-    -processorpath "ci/$JAVA_ERROR_PRONE:ci/$DATAFLOW_SHADED:ci/$JFORMAT_STRING" \
+    -processorpath "ci/$java_error_prone:ci/$dataflow_shaded:ci/$jformat_string" \
     '-Xplugin:ErrorProne -Xep:CatchingUnchecked:ERROR' \
     -classpath "$flutter_bin/cache/artifacts/engine/android-x64/flutter.jar" \
     $temp_dir/Pigeon.java; then
     echo "javac $temp_dir/Pigeon.java failed"
     exit 1
   fi
 
-  dartfmt -w $temp_dir/pigeon.dart
-  dartanalyzer $temp_dir/pigeon.dart --fatal-infos --fatal-warnings --packages ./e2e_tests/test_objc/.packages
-
   rm -rf $temp_dir
 }
 
 # test_null_safe_dart(<path to pigeon file>)
 #
 # Compiles the pigeon file to a temp directory and attempts to run the dart
-# analyzer on it with null safety turned on.
-test_null_safe_dart() {
-  temp_dir=$(mktmpdir)
+# analyzer on it with and without null safety turned on.
+test_pigeon_dart() {
+  echo "test_pigeon_dart($1)"
+  temp_dir_1=$(mktmpdir)
+  temp_dir_2=$(mktmpdir)
 
   pub run pigeon \
     --input $1 \
-    --dart_out $temp_dir/pigeon.dart
+    --dart_out $temp_dir_1/pigeon.dart &
+  null_safe_gen_pid=$!
 
-  dartanalyzer $temp_dir/pigeon.dart --fatal-infos --fatal-warnings --packages ./e2e_tests/test_objc/.packages
-  rm -rf $temp_dir
+  pub run pigeon \
+    --no-dart_null_safety \
+    --input $1 \
+    --dart_out $temp_dir_2/pigeon.dart &
+  non_null_safe_gen_pid=$!
+
+  wait $null_safe_gen_pid
+  wait $non_null_safe_gen_pid
+
+  dartanalyzer $temp_dir_1/pigeon.dart --fatal-infos --fatal-warnings --packages ./e2e_tests/test_objc/.packages &
```

Comment:
what's ./e2e_tests/test_objc/.packages about? Add a comment?

Reply:
Commented, I could probably make flutter a dev dependency for pigeon as an alternative.